File: Algorytmika/zadanie-3/main.py
def biggest_k(first_string, second_string):
    l = min(len(first_string), len(second_string))
    
    for i in range(l, 0, -1):
        if first_string[:i] == second_string[-i:]:
            return i
    return 0

x = "psikacz"
y = "mopsik"
print(biggest_k(x, y))

File: Algorytmika/zadanie-3/main.c
#include <stdio.h>
#include <string.h>

int biggest_k(const char* string1, const char* string2){
    int len_x = strlen(string1);
    int len_y = strlen(string2);
    int len = len_x > len_y ? len_y : len_x;

    for (int k = len; k > 0; k--) {
        int match = 1;
        int i;
        for (i = 0; i < k; i++) {
            if (string1[i] != string2[len_y - k + i]) {
                match = 0;
                break;
            }
        }
        if (match) {
            return k;
        }
    }
    return 0;
}

int main(){
    printf("%d\n", biggest_k( "psikacz","mopsik"));
}